module-4/backend/week_four.md

```diff
@@ -5,18 +5,33 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+  Not yet started, started, finished, delivered, accepted/rejected
 * What are the four types of cards available in Pivotal Tracker?
+  Feature, chore, bug, release
 * What's your favorite project management tool?
+  I'm a fan of Waffle because of it's simplicity, ease of use, and GitHub integrations, but Pivotal Tracker is pretty interesting
+  even if just for the amount of information you can see in the analytics pieces it offers.
 * Why do we create staging environments?
+  Staging environments are great because they allow for a place to test that your code works in a production environment. They are also useful for having a set of new features live and ready to go for a push to production.
 * What are the characteristics of a good README (in your opinion)?
+  A good README should be concise, use language which is easily understood, and address the appropriate issue for the project. If it's say a gem then it should include a walk-through of how to utilize it's various features, and if it is an open project it should contain guidelines for contribution.
 * What's one main improvement you're going to make to your code regarding accessibility issues?
+  I'd like to focus more on using HTML tags which are conducive to a screen reader and tabbing.
 * What are some basic security concerns to be aware of when building applications?
+  Knowing how to appropriately store passwords(hashed vs raw). Being aware of basic techniques such as strong params and proper SQL method writing to prevent the passing in of harmful data.
 * Why is continuous integration helpful/important?
+  CI is helpful because it runs your test suite to ensure that all of the applications tests are currently passing. It also 
+  can give insight as to where problems are arising, sometimes suggesting whose fault it might be(neat but scary feature). It is   important because it makes certain that an application is functional prior to deployment.
 * What are some continuous integration tools?
+  Travis, Jenkins, and Circle are the ones that I remember.
 
 #### Review  
 
 * What are some characteristics of "good" git workflow?
+  Committing at an appropriate interval such that good work is not lost; thorough commit messages such that both the developer and their team are aware of what each commit contains; tagging team members in pull request conversations to discuss any questionable portions of the code.
 * What are the four fundamental concepts of object oriented programming?
+  Inheritance, encapsulation, abstraction, and polymorphism
 * What's a module in Ruby and what's the difference between a class and a module?
+  A module in Ruby is like a set of methods and possibly constants that can be inherited by/ assigned to a class. A class is a structure that allows for the storage of information with attributes as well as manipulation of that information and interactions with other objects via its defined methods; an instance of a class may be created; however, a module cannot.
```

Comment:
a class does not inherit from a module. a module is mixed in to a class - it's "included." when it's included, we have access to those methods defined within the module within the class itself.